x/peggy/keeper/keeper.go

```diff
@@ -225,33 +225,31 @@ func (k *Keeper) GetLastUnbondingBlockHeight(ctx sdk.Context) uint64 {
 	return types.UInt64FromBytes(bytes)
 }
 
-// GetUnslashedValsets returns all the unslashed validator sets in state
-func (k *Keeper) GetUnslashedValsets(ctx sdk.Context, maxHeight uint64) (out []*types.Valset) {
+// GetUnslashedValsets returns all the "ready-to-slash" unslashed validator sets in state (valsets at least signedValsetsWindow blocks old)
+func (k Keeper) GetUnslashedValsets(ctx sdk.Context, signedValsetsWindow uint64) (out []*types.Valset) {
 	lastSlashedValsetNonce := k.GetLastSlashedValsetNonce(ctx)
-
-	k.IterateValsetBySlashedValsetNonce(ctx, lastSlashedValsetNonce, maxHeight, func(_ []byte, valset *types.Valset) bool {
-		if valset.Nonce > lastSlashedValsetNonce {
+	blockHeight := uint64(ctx.BlockHeight())
+	k.IterateValsetBySlashedValsetNonce(ctx, lastSlashedValsetNonce, func(_ []byte, valset *types.Valset) bool {
+		// Implicitly the unslashed valsets appear after the last slashed valset,
+		// however not all valsets are ready-to-slash since validators have a window
```

Comment:
```suggestion
		// Implicitly, the unslashed valsets appear after the last slashed valset,
		// however, not all valsets are ready-to-slash since validators have a window.
```